src/test/kotlin/interactor/GetCityHasMoreSavingsPerMonthInteractorTest.kt

```diff
@@ -19,225 +19,168 @@ internal class GetCityHasMoreSavingsPerMonthInteractorTest {
-            transportationsNullAndFoodPriceAndApartmentAndAverageSalaryNotNull.realEstatesPrices
-        }returns RealEstatesPrices(10f,20f,30f,40f,10f,50f)
+            cityEntityPrices.realEstatesPrices
+        }returns RealEstatesPrices(null,20f,30f,40f,10f,50f)
 
         every {
-            transportationsNullAndFoodPriceAndApartmentAndAverageSalaryNotNull.averageMonthlyNetSalaryAfterTax
+            cityEntityPrices.averageMonthlyNetSalaryAfterTax
         }returns 200f
 
-        every {
-            transportationsNullAndFoodPriceAndApartmentAndAverageSalaryNotNull.transportationsPrices
-        }returns TransportationsPrices(null,null,null,null,null,null)
-
         every {
             dataSource.getAllCitiesData()
-        } returns listOf(transportationsNullAndFoodPriceAndApartmentAndAverageSalaryNotNull)
-
+        } returns listOf(cityEntityPrices)
 
 
         // when find the most  city in the world have more savings per month.
         citySavings = GetCityHasMoreSavingsPerMonthInteractor(dataSource)
         val getCity = citySavings.execute()
 
 
-        // done
-        assertEquals(transportationsNullAndFoodPriceAndApartmentAndAverageSalaryNotNull, getCity)
+        // then
+        assertEquals(cityEntityPrices, getCity)
 
 
     }
     @Test
-    fun should_ReturnNull_When_FoodPriceNull() {
+    fun should_ThrowException_When_FoodPriceNull() {
 
         // given city with food price is null
         every {
-            foodPricesWithNull.foodPrices
-        } returns FoodPrices(10f,20f,null,40f,10f,50f)
-
-        every {
-            foodPricesWithNull.realEstatesPrices
-        } returns RealEstatesPrices(10f,20f,null,40f,10f,50f)
+            cityEntityPrices.foodPrices
+        } returns FoodPrices(null,20f,null,40f,10f,50f)
 
         every {
             dataSource.getAllCitiesData()
-        } returns listOf(foodPricesWithNull)
+        } returns listOf(cityEntityPrices)
+
 
         // when find the most  city in the world have more savings per month.
         citySavings = GetCityHasMoreSavingsPerMonthInteractor(dataSource)
-        val getNull = citySavings.execute()
+        val savingCityExecutable = Executable { citySavings.execute()}
+
 
 
-        // done
-        assertNull(getNull)
+        // then it throws an exception
+        assertThrows(Exception::class.java,savingCityExecutable)
 
 
     }
     @Test
-    fun should_ReturnNull_When_ApartmentNull() {
+    fun should_ThrowException_When_ApartmentNull() {
 
         // given city with apartment is null
         every {
-            apartmentPricesWithNull.foodPrices
+            cityEntityPrices.foodPrices
         }returns  FoodPrices(10f,20f,30f,15f,25f,40f)
 
         every {
-            apartmentPricesWithNull.realEstatesPrices
+            cityEntityPrices.realEstatesPrices
         } returns RealEstatesPrices(10f,20f,null,null,10f,15f)
 
+        every {
+            cityEntityPrices.averageMonthlyNetSalaryAfterTax
+        }returns null
```

Comment:
when you're testing the case of null apartment, averageMonthlyNetSalaryAfterTax should not be null.

Reply:
done!